packages/services/f-analytics/src/mixins/analytics.mixin.vue

```diff
@@ -1,5 +1,166 @@
+            }
+        },
+
+        prepareServersideAnalytics () {
+            if (this.isServerSide) {
+                // Only available serverside
+                const platformData = { ...this.platformData };
+
+                platformData.environment = process.env.justEatEnvironment || 'localhost';
+                platformData.version = process.env.FEATURE_VERSION || '0.0.0.0';
+                platformData.instancePosition = process.env.INSTANCE_POSITION || 'N/A';
+
+                // Is of type `httponly` so need to read serverside
+                platformData.jeUserPercentage = this.$cookies.get('je-user_percentage') || 0;
+
+                this.updatePlatformData(platformData);
+            }
+        },
+
+        preparePage () {
+            if (!window.dataLayer) {
+                const queryString = (this.getGtmSettings.auth !== undefined)
+                    ? `&gtm_auth=${this.getGtmSettings.auth}&gtm_preview=${this.getGtmSettings.preview}&gtm_cookies_win=${this.getGtmSettings.cookiesWin}`
+                    : '';
+
+                // See : https://developers.google.com/tag-manager/quickstart
+                const headJsGtmTag = `(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+                new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+                j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+                'https://www.googletagmanager.com/gtm.js?id='+i+dl${queryString};f.parentNode.insertBefore(j,f);
+                })(window,document,'script','dataLayer','${this.getGtmSettings.id}');`;
+                const script = document.createElement('script');
+                script.text = headJsGtmTag;
+                document.head.prepend(script);
+
+                const iframeTag = document.createElement('iframe');
+                iframeTag.src = `https://www.googletagmanager.com/ns.html?id=${this.getGtmSettings.id}`;
+                iframeTag.setAttribute('height', 0);
+                iframeTag.setAttribute('width', 0);
+                iframeTag.style.display = 'none';
+                iframeTag.style.visibility = 'hidden';
+                const noScript = document.createElement('noscript');
+                noScript.appendChild(iframeTag);
+                document.body.prepend(noScript);
+            }
+        },
+
+        prepareAnalytics () {
+            const platformData = { ...this.platformData };
+
+            platformData.name = this.$route.name;
+            platformData.appType = 'web';
+            platformData.applicationId = 7;
+            platformData.userAgent = navigator.userAgent || 'N/A';
+            platformData.branding = this.getBrand(this.$i18n.locale);
+            platformData.country = this.getCountry(this.$i18n.locale);
+            platformData.language = this.getLanguage(this.$i18n.locale);
+            platformData.currency = this.getCurrency(this.$i18n.locale);
```

Comment:
I _think_ lines 93 -102 can be re-written to something like 
```
const platformData = { 
           ...this.platformData,
           name: this.$route.name,
           appType: 'web',
           (same with the rest of these changes)
  };

Reply:
I tried that before but if any of the lines failed for some reason the error message only indicates the first line and not the line that caused the error.  So for maintenance I reasons I did it this way.